src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Nls.cs

```diff
@@ -74,15 +74,15 @@ private int NlsGetLocaleInfo(LocaleNumberData type)
 
             // Ask OS for data, note that we presume it returns success, so we have to know that
             // sWindowsName is valid before calling.
-            Debug.Assert(_sWindowsName != null, "[CultureData.DoGetLocaleInfoInt] Expected _sWindowsName to be populated by already");
-            return GetLocaleInfoExInt(_sWindowsName, lctype);
+            Debug.Assert(_sRealName != null, "[CultureData.DoGetLocaleInfoInt] Expected _sRealName to be populated by already");
```

Comment:
nit:
```suggestion
            Debug.Assert(_sRealName != null, "[CultureData.DoGetLocaleInfoInt] Expected _sRealName to be populated already");
```
(and few other places in this file)